jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py

```diff
@@ -148,271 +164,202 @@ def user_api(self):
-            # a 'sb' message for mouseover info. this should be removed when
-            # unit change messaging is improved and is a temporary fix
-            self.hub.broadcast(GlobalDisplayUnitChanged('sb',
-                                                        self.flux_unit.selected,
-                                                        sender=self))
-
-            if not self.flux_unit.selected:
-                y_display_unit = self.spectrum_viewer.state.y_display_unit
-                flux_unit_str = str(u.Unit(y_display_unit * y_unit_solid_angle))
-                self.flux_unit.selected = flux_unit_str
-
-    @observe('spectral_unit_selected')
-    def _on_spectral_unit_changed(self, *args):
-        xunit = _valid_glue_display_unit(self.spectral_unit.selected, self.spectrum_viewer, 'x')
-        if self.spectrum_viewer.state.x_display_unit != xunit:
-            self.spectrum_viewer.state.x_display_unit = xunit
-            self.hub.broadcast(GlobalDisplayUnitChanged('spectral',
-                               self.spectral_unit.selected,
-                               sender=self))
+    @property
+    def spectral_y_unit(self):
+        return self.sb_unit_selected if self.spectral_y_type_selected == 'Surface Brightness' else self.flux_unit_selected  # noqa
 
-    @observe('spectral_y_type_selected')
-    def _on_spectral_y_type_selected(self, msg):
+    def _on_add_data_to_viewer(self, msg):
+        # toggle warning message for cubes without PIXAR_SR defined
+        if self.config == 'cubeviz':
+            # NOTE: this assumes data_collection[0] is the science (flux/sb) cube
+            if (
+                len(self.app.data_collection) > 0
+                and not self.app.data_collection[0].meta.get('PIXAR_SR')
+            ):
+                self.pixar_sr_exists = False
+
+        viewer = msg.viewer
+
+        if (not len(self.spectral_unit_selected)
+                or not len(self.flux_unit_selected)
+                or not len(self.angle_unit_selected)
+                or (self.config == 'cubeviz' and not len(self.spectral_y_type_selected))):
+
+            data_obj = msg.data.get_object()
+            if isinstance(data_obj, Spectrum1D):
+
+                self.spectral_unit._addl_unit_strings = self.spectrum_viewer.state.__class__.x_display_unit.get_choices(self.spectrum_viewer.state)  # noqa
+                if not len(self.spectral_unit_selected):
+                    try:
+                        self.spectral_unit.selected = str(data_obj.spectral_axis.unit)
+                    except ValueError:
+                        self.spectral_unit.selected = ''
+
+                angle_unit = check_if_unit_is_per_solid_angle(data_obj.flux.unit, return_unit=True)
+                flux_unit = data_obj.flux.unit if angle_unit is None else data_obj.flux.unit * angle_unit  # noqa
+
+                if not self.flux_unit_selected:
+                    if flux_unit in (u.count, u.DN):
```

Comment:
electron / s too?

Reply:
For now, this should be anything that we support that is not given as an option by `create_flux_equivalencies_list(u.Jy, u.Hz)`, so I either could add `electron /s` (and any others we know) here or switch this to `if flux_unit not in self.flux_unit.choices`.  I think eventually we'll want this logic to be a little more reactive to the data, but hopefully this at least gives basic support.